bzzeth/bzzeth_test.go

```diff
@@ -223,3 +260,170 @@ func isPeerDisconnected(id enode.ID, b *BzzEth) (p *Peer) {
 	}
 	return
 }
+
+func newBlockHeaderExchange(tester *p2ptest.ProtocolTester, peerID enode.ID, requestID uint32, offered *NewBlockHeaders, wanted [][]byte) error {
+	return tester.TestExchanges(
+		p2ptest.Exchange{
+			Label: "NewBlockHeaders",
+			Triggers: []p2ptest.Trigger{
+				{
+					Code: 1,
+					Msg:  offered,
+					Peer: peerID,
+				},
+			},
+			Expects: []p2ptest.Expect{
+				{
+					Code: 2,
+					Msg: GetBlockHeaders{
+						ID:     requestID,
+						Hashes: wanted,
+					},
+					Peer: peerID,
+				},
+			},
+		})
+}
+
+func blockHeaderExchange(tester *p2ptest.ProtocolTester, peerID enode.ID, requestID uint32, wantedData []rlp.RawValue) error {
```

Comment:
this does not seem to test much

Reply:
blockHeaderExchange does not test anything from the protocol response perspective, but TestNewBlockHeaders testcase checks the effect of receiving this message.  Especially see the functions checkStorage and checkDelivery.